public/app/features/plugins/AppRootPage.tsx

```diff
@@ -80,6 +79,14 @@ class AppRootPage extends Component<Props, State> {
       return <div>No Root App</div>;
     }
 
+    // When no naviagion is set, give full control to the app plugin
+    if (!nav) {
+      if (plugin && plugin.root) {
+        return <plugin.root meta={plugin.meta} query={query} path={path} onNavChanged={this.onNavChanged} />;
+      }
+      return <PageLoader />;
```

Comment:
@besartberisha is this loading behavior what you want?

Reply:
Hey @ryantxu yes exactly, far more cleaner solution rather than exposing different callbacks, thank you.